--- system/src/stm32f4-hal/stm32f4xx_hal_sd.c	2015-03-22 15:25:59.820077169 -0400
+++ system/src/stm32f4-hal/stm32f4xx_hal_sd.c	2015-03-22 15:27:07.959305612 -0400
@@ -1605,7 +1605,7 @@
     /* Byte 10 */
     tmp = (uint8_t)((hsd->CSD[2] & 0x0000FF00) >> 8);
     
-    pCardInfo->CardCapacity  = ((pCardInfo->SD_csd.DeviceSize + 1)) * 512 * 1024;
+    pCardInfo->CardCapacity  = ((uint64_t)(pCardInfo->SD_csd.DeviceSize + 1)) * 512 * 1024;
     pCardInfo->CardBlockSize = 512;    
   }
   else
